src/core/lib/surface/server.cc

```diff
@@ -1599,13 +1589,15 @@ void grpc_server_shutdown_and_notify(grpc_server* server,
   gpr_mu_unlock(&server->mu_global);
 
   /* Shutdown listeners */
-  for (l = server->listeners; l; l = l->next) {
-    GRPC_CLOSURE_INIT(&l->destroy_done, listener_destroy_done, server,
-                      grpc_schedule_on_exec_ctx);
-    l->destroy(server, l->arg, &l->destroy_done);
-    if (server->channelz_server != nullptr && l->socket_uuid != 0) {
-      server->channelz_server->RemoveChildListenSocket(l->socket_uuid);
+  for (auto& listener : server->listeners) {
+    auto* channelz_listen_socket_node =
```

Comment:
Can we avoid auto since this isn't just being used as a pointer but is also being dereff'ed below?

Reply:
Done.